docs/advanced/controlled-inputs.md

```diff
@@ -1,7 +1,68 @@
 # Controlled Inputs
 
+## set
+
+To change values in Leva store from the outside, use the function API by passing `() => schema` to `useControls`.
+
+```jsx
+const [{ text }, set] = useControls(() => ({ text: 'my string' }))
+
+return <input type="text" value={text} onChange={(e) => set({ text: e.target.value })} />
+```
+
+Use `set` to update values from outside of Leva GUI.
+
+See an [example in Storybook](https://leva.pmnd.rs/?path=/story/misc-controlled-inputs--external-updates-with-set).
+
+## onChange
+
+`onChange` callback in the schema is called for all changes to the value. Values with `onChange` will no longer cause React to rerender (unless passing an additional `transient` flag set to `true`, see below), so you can use it to efficiently update frequently changing values.
+
+```jsx
+const divRef = React.useRef(null)
+const data = useControls({
+  color: {
+    value: '#f00',
+    onChange: (v) => {
+      // imperatively update the world after Leva input changes
+      divRef.current.style.color = v
+    },
+  },
+})
+
+// `data.color` is undefined
+```
+
+See an [example in Storybook](https://leva.pmnd.rs/?path=/story/misc-input-options--on-change).
+
+### Transient
+If you need the `onChange` callback while 
+
+## set and onChange
+
+With `set` and `onChange` we can bind to any imperative API. Whenever external
+
+```jsx
+const [, set] = useControls(() => ({
+  position: {
+    value: { x: 0, y: 0 },
+    onChange: (value) => {
+      // imperatively update the world after Leva input changes
+    },
+  },
+}))
```

Comment:
```suggestion
}))

const targetRef = useRef()
useDrag(
  ({ offset: [x, y] }) => set({ position: { x, y } }),
  { domTarget: targetRef }
)
```

I wanted to find the simplest example that would justify two-way imperative binding. I guess 2D dragging fits.